### ISSUE TITLE
v10.1.0 breaks keyboard navigation in Max Mega Menu

### ISSUE THREAD

**Issue by megamenu** [woothemes/woocommerce#61444]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Max Mega Menu allows users to navigate the menu using only a keyboard. Tab can be used to navigate through items, Enter can be used to open or close a sub menu. Specifically, a change introduced in v10.1.0 breaks the ability for a user to use the Enter key to open or close a sub menu. It works as expected in v10.0.4 and below.

I imagine it is due to this code in woocommerce.js:

```
`	// Global accessibility handler for all links with role="button"
	// This ensures both spacebar and enter keypresses trigger click events, as per ARIA specification
	document.body.addEventListener( 'keydown', function ( event ) {
		if ( ! event.target.matches( 'a[role="button"]' ) ) {
			return;
		}

		if ( event.key === ' ' || event.key === 'Enter' ) {
			event.preventDefault();
			event.target.click();
		}
	} );
`
```

Please ensure you only target elements that WooCommerce is directly responsible for. I don't think WooCommerce should be globally targeting elements like this, it is not its responsibility to do so.

### Expected behavior

Sub menus open when enter key is pressed.

### Actual behavior

Nothing happens when enter key is pressed.

### Steps to reproduce

Install Max Mega Menu
Create a simple menu with a sub menu
Use the tab key to navigate to the parent item and press enter
Sub menu should appear, but doesn't.
Disable WooCommerce, try again, sub menu appears.

### WordPress Environment

Any.

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by sunyatasattva** [comment]
Hey there, thanks for the report. I think you've got a point. @kmanijak what do you think, as the author of that PR? I see [a comment](https://github.com/woocommerce/woocommerce/pull/59578#pullrequestreview-3005609131) by @Aljullu warning about this.

I tend to agree with @megamenu here, we shouldn't enforce this behavior on everyone else, even though this is the correct a11y practice.

**Comment by kmanijak** [comment]
> Hey there, thanks for the report. I think you've got a point. @kmanijak what do you think, as the author of that PR? I see https://github.com/woocommerce/woocommerce/pull/59578#pullrequestreview-3005609131 by @Aljullu warning about this.

Hmm.. I see. My motivation [was the original issue](https://github.com/woocommerce/woocommerce/issues/59542) from accessibility experts suggested we should actually target all the links with role button:

> The implementation should be broad enough that it adds spacebar support to all links with a role of button, even those created by extensions.

I was thinking that maybe we could find a middle ground in which we keep the accessibility change and make sure it's not disruptive to the plugin so I tried reproducing the issue and I don't think I can. Does menu require specific setting in which the behaviour breaks? This is vide of me using Enter and Spacebar to open submenu, unless I misunderstood the problem 🙌 

https://github.com/user-attachments/assets/b7cd27a0-97fc-47a9-9e47-2463291695b6

---

Last resort would be probably to revert the PR and target only known places as I don't think we can sustainably target only anchors coming from core.

**Comment by megamenu** [comment]
@kmanijak the issue you've linked to refers to being able to use _spacebar_ to 'click' a button. However the code added targets both Spacebar and Enter.

In either case I do not think WooCommerce should be taking on this responsibility globally. Other plugins need the option to handle keyboard interactivity in their own way. The code should be restricted to targeting only elements created by WooCommerce (`.woocommerce a[role="button"]`) or at the very least some way of plugins opting out (`a[role="button"]:not(.custom-class-plugin-authors-can-add`) (this would be least preferable).

To recreate the behaviour, go to Appearance > Menus, hover over the Checkout menu item and click the blue mega menu button. In the settings, enable the Disable Link option and Save. Does that recreate the behaviour?

**Comment by kmanijak** [comment]
I see, I could reproduce the problem and I agree it's too aggressive to add a listener like this.

> The code should be restricted to targeting only elements created by WooCommerce (.woocommerce a[role="button"])

Unfortunately, that would also target Max Mega Menu (or any other content within the scope of class). There's no reliable way of extracting WooCommerce core anchors from external content if the content is mixed. I think I'll revert the PR and just add an event handler to the places that were recognised as problematic.

cc: @PanosSynetos, letting you know that Composite Products would have to add their own event handler for anchor with button role (`a['role=button']`) that we were addressing in this PR: https://github.com/woocommerce/woocommerce/pull/59578.

**Comment by amberhinds** [comment]
I wonder if instead of reverting this it should be improved to not conflict with Max Mega Menu.

It seems like there could be an issue because this code in Woo was triggered on key down instead of keyup. Buttons activate on Space keyup (not keydown). Could that be causing issues?
